### DISCUSSION TITLE
Consider a general term for dereferencing a URL name

### DISCUSSION THREAD

**Discussion by igbanam** [martenframework/marten#246] [Ideas]
Right now, in templates, URL names are dereferenced using `{% url "route:to:dereference" %}`. In handlers, what seems like the same functionality is called `reverse("route:to:dereference")`.

Could Marten consider changing `reverse` to `url` for the handlers as well? 

https://github.com/martenframework/marten/blob/4ae28f0f58c7758722a8c9f35ac72703cad51fa4/src/marten/routing/reverser.cr#L20

**Comment by ellmetha** [comment]
Hey! 👋 

The "Reverse URL resolution" concept is important to the framework and is [heavily documented](https://martenframework.com/docs/handlers-and-http/routing#reverse-url-resolutions). The rationale behind this naming is that it performs the reverse operation of the route pattern matching process (where a route/handler association is resolved based on a given URL path). As such, given a route name and optional parameters, the `#reverse` method constructs the corresponding route if one exists. Marten uses this naming internally for its routing component (eg. [`Marten::Routing::Map#reverse `method](https://martenframework.com/docs/api/0.5/Marten/Routing/Map.html#reverse(name%3AString%7CSymbol%2Cparams%3AHash(String%7CSymbol%2CParameter%3A%3ATypes))-instance-method)).

Additionally, handlers and templates are components that may not necessarily be used by the same audience: handlers will likely be written by Crystal developers while templates could be written by frontend developers that may not be familiar with the "reverse URL resolution" concept. That's why the `url` naming was chosen for the template tag corresponding to the `#reverse` method.

For these reasons, I don't think the `reverse` naming and its associated concept should go away. That said, I am open to introducing a `#url` alias for `#reverse` in handlers (or a `reverse` tag alias for templates). How does that sound?

**Reply by igbanam** [reply]
This alias is more what I was leaning towards, to be honest. This isn't a functionality change per se. It's just sugar to make a marten codebase seem more homogenous.

**Reply by ellmetha** [reply]
Added in 73778f82242439c52ec6504c9c9b0297bffb6e5b. 👌

**Reply by igbanam** [reply]
Fantastic. Thank you!
